### ISSUE TITLE
Package warning if kernel extension was not loaded

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/Semigroups#537]
@ManuelAFDelgado is visiting me right now, and we stumbled over the following:

The `semigroups` package has this in its `PackageInfo.g`:
```gap
AvailabilityTest := function()
  local semigroups_so;
  semigroups_so := Filename(DirectoriesPackagePrograms("semigroups"),
                            "semigroups.so");
  if (not "semigroups" in SHOW_STAT()) and semigroups_so = fail then
    Info(InfoWarning, 1, "Semigroups: the kernel module is not compiled, ",
         "the package cannot be loaded.");
    return fail;
  fi;
  return true;
end,
```

So it uses `Info(InfoWarning`. The problem with that is that if one wants to write code (e.g. tests) which only is run when `semigroups` is tested, it is now difficult to do so w/o triggering that warning. In particular:
```
gap> TestPackageAvailability("semigroups");
#I  Semigroups: the kernel module is not compiled, the package cannot be loaded.
#I  Semigroups: the kernel module is not compiled, the package cannot be loaded.
fail
```
One can of course work around that, by temporarily changing `InfoWarning` to level 0 and then restoring it afterwards, but that's a bit cumbersome.

Future GAP versions may also have an `IsPackageLoaded` function that does the right thing, so hopefully this won't be really an issue down the road.

But the reason I bring this up anyway is that strictly speaking, this use goes against the "documentation" for `PackageInfo.g` in the Example package, which states this:
```gap
##  Provide a test function for the availability of this package.
##  For packages containing nothing but GAP code, just say 'ReturnTrue' here.
##  For packages which may not work or will have only partial functionality,
##  use 'LogPackageLoadingMessage( PACKAGE_WARNING, ... )' statements to
##  store messages which may be viewed later with `DisplayPackageLoadingLog'.
##  Do not call `Print' or `Info' in the `AvailabilityTest' function of the 
##  package.
```
So this issue could be fixed on the `semigroups` side by using `LogPackageLoadingMessage( PACKAGE_WARNING, ...)` instead of `Info(InfoWarning, ...)`.

(That said, my guess is that perhaps you added this `Info` on purpose, to get better feedback quickly; well, I personally also am not that happy about the state of how and which warning messages we print upon `LoadPackages`, so I hope this can also be improved at some point... *sigh*)

**Comment by james-d-mitchell** [comment]
Thanks for the report @fingolfin, I'll be happy to have Semigroups behave correctly w.r.t. the documentation.

**Comment by ManuelAFDelgado** [comment]
The purpose of this comment is to observe that loading a non compiled version of a package that should be compiled may lead to wrong results... Max (@fingolfin), this is just another evidence that the "fail" you mentioned must really be replaced by "false"...

The examples below were produced in three different GAP sessions, with the packages loaded as indicated:
#Example1# without loading any package (apart from those that are loaded automatically)
## a monoid -- poi3 is an inverse monoid
gap> g0:=Transformation([4,1,2,4]);;
gap> g1:=Transformation([1,3,4,4]);;
gap> g2:=Transformation([2,4,3,4]);;
gap> poi3:= Monoid(g0,g1,g2);;
gap> 
gap> DC := GreensDClasses(poi3);;
gap> DisplayEggBoxOfDClass(DC[3]);
[ [  0,  0,  1 ],
  [  1,  0,  0 ],
  [  0,  1,  0 ] ]


#Example2# loading the package "semigroups", the *NON* COMPILED VERSION
LoadPackage("semigroups"); 

## a monoid -- poi3 is an inverse monoid
gap> g0:=Transformation([4,1,2,4]);;
gap> g1:=Transformation([1,3,4,4]);;
gap> g2:=Transformation([2,4,3,4]);;
gap> poi3:= Monoid(g0,g1,g2);;
gap> 
gap> DC := GreensDClasses(poi3);;
gap> DisplayEggBoxOfDClass(DC[3]);
[ [  0,  0,  1 ],
  [  0,  0,  1 ],
  [  0,  0,  1 ] ]
##WRONG!!!!!!!!! (poi3 is an inverse monoid)


#Example3# Now loading the compiled version of the package "semigroups"
LoadPackage("semigroups"); 

## a monoid -- poi3 is an inverse monoid
gap> g0:=Transformation([4,1,2,4]);;
gap> g1:=Transformation([1,3,4,4]);;
gap> g2:=Transformation([2,4,3,4]);;
gap> poi3:= Monoid(g0,g1,g2);;
gap> 
gap> DC := GreensDClasses(poi3);;
gap> DisplayEggBoxOfDClass(DC[3]);
[ [  0,  0,  1 ],
  [  0,  1,  0 ],
  [  1,  0,  0 ] ]

**Comment by james-d-mitchell** [comment]
Resolved in v3.0.20.
